cmd/minikube/cmd/delete.go

```diff
@@ -601,49 +602,99 @@ func killMountProcess() error {
 	return nil
 }
 
+// killProcess takes a path to look for a pidfile(space-separated),
+// it reads the file and converts it to a bunch of pid ints,
+// then it tries to kill each one of them.
+// If no errors were ecnountered, it cleans the pidfile
 func killProcess(path string) error {
 	pidPath := filepath.Join(path, constants.MountProcessFileName)
 	if _, err := os.Stat(pidPath); os.IsNotExist(err) {
 		return nil
 	}
-
 	klog.Infof("Found %s ...", pidPath)
-	out, err := os.ReadFile(pidPath)
+
+	ppp, err := GetPids(pidPath)
 	if err != nil {
-		return errors.Wrap(err, "ReadFile")
+		return err
 	}
-	klog.Infof("pidfile contents: %s", out)
-	pid, err := strconv.Atoi(string(out))
-	if err != nil {
-		return errors.Wrap(err, "error parsing pid")
+
+	// we're giving a change to each pid, to be killed
+	var errs []error
+	for _, pp := range ppp {
+		err := tryKillOne(pp)
+		if err != nil {
+			errs = append(errs, err)
+		}
+
+	}
+
+	if len(errs) == 1 {
+		// if we've encountered only one error, we're returning it:
+		return errs[0]
+	} else if len(errs) != 0 {
+		// if multiple errors were encountered, we're outputting them
+		// and returning a new multiple-errors error:
+		out.Step(style.Failure, "Ecountered multiple errors:")
+		for _, e := range errs {
+			out.Err("%v\n", e)
+		}
+		return errors.New("multiple errors encountered while closing mount processes")
 	}
-	// os.FindProcess does not check if pid is running :(
+
+	// if no errors were encoutered, it is safe to clean out the pidFile
+	if err := os.Remove(pidPath); err != nil {
+		return errors.Wrap(err, "While closing mount-pids file")
+	}
+
+	return err
+}
+
+// tryKillOne takes a PID as argument and tries to SIGKILL it
+func tryKillOne(pid int) error {
```

Comment:
This name is a bit odd, I'd recommend something like `killProcess` maybe?

Reply:
Unfortunately this is a helper function of the killProcess function :/
What about something like `trySignalProcess`?